core/ledger/pvtdatastorage/retroactive_hashed_index.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pvtdatastorage
+
+import (
+	"github.com/hyperledger/fabric-protos-go/ledger/rwset"
+	"github.com/hyperledger/fabric/common/ledger/util/leveldbhelper"
+
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/rwsetutil"
+	"github.com/hyperledger/fabric/core/ledger/util"
+	"github.com/pkg/errors"
+)
+
+const (
+	previousDataVersion = ""
+	currentDataVersion  = "2.6"
+	maxUpgradeBatchSize = 4 * 1024 * 1024 // 4 MB
+)
+
+func checkAndConstructHashedIndex(storePath string, ledgerIDs []string) error {
+	i, err := leveldbhelper.RetrieveDataFormatInfo(storePath)
```

Comment:
I personally find it harder to read. I would like to be able to understand a line of code without having to look elsewhere to remind me what a single character actually is, single character variables I think is one of the worst ideas to come out of idiomatic go

Reply:
If it is a well known noun like an "index" I can agree to an i. Same for k for "key" and v for "value" since they are well known nouns in the domain. But an "Info" is a one-off that nobody reading ledger code knows about, and therefore deserves more than a letter (especially a letter that usually means something else).

Reply:
@manish-sethi It is not as black and white as you indicate. You can't just assume that other viewpoints are merely opinions or non-adapted developers while your own viewpoints are the correct ones, and that `i` is more readable than `info`.

While it is true that variables in close proximity should be short or even single-letter, there are also suggestions in the same Go guidelines that `i` shouldn't be used for things that aren't loop indexes/incrementers.

https://github.com/golang/go/wiki/CodeReviewComments#variable-names

> Common variables such as loop indices and readers can be a single letter (i, r). More unusual things and global variables need more descriptive names.

In the context of ledger, an Info is a more unusual thing (not a well-known noun as I mentioned earlier).

https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html

> Finally, certain single letter variables have traditionally been associated with loops and counting. For example, i, j, and k are commonly the loop induction variable for simple for loops. [2] n is commonly associated with a counter or accumulator. v is a common shorthand for a value in a generic encoding function, k is commonly used for the key of a map. a and b are generic names for parameters comparing two variables of the same type. x and y are generic names for local variables created for comparison, and s is often used as shorthand for parameters of type string.
> 
> As with the db example above programmers expect i to be a loop induction variable. If you ensure that i is always a loop variable, not used in other contexts outside a for loop. When readers encounter a variable called i, or j, they know that a loop is close by.

This debate has already gone on longer than it should have, it was a minor comment and I'm happy to move on without it. My larger point is that we should try to be receptive to comments from people reading the code (especially when they are backed up by Go guidelines!). Code readability is best judged by people trying to read the code. I actually didn't know about the written Go guidelines about not using `i` outside of loop context due to potential confusions, I only knew that it tripped ME up when first scanning the code for that very reason. If there are two people indicating the readability could be improved, isn't that enough?

Reply:
> @manish-sethi It is not as black and white as you indicate. You can't just assume that other viewpoints are merely opinions or from non-adapted developers while your own viewpoints are the correct ones, and that i is more readable than info.

@denyeart - Not sure why you are giving this a spin. All I am saying is that we all need to get over our preferences/view points and try to get use to the convention of the language. E.g., if you apply these golang convention to java code, you would hear other way around. I'll paste the complete text from where you pasted selected text.

> Variable names in Go should be short rather than long. This is especially true for local variables with limited scope. Prefer c to lineCount. Prefer i to sliceIndex.

> The basic rule: the further from its declaration that a name is used, the more descriptive the name must be. For a method receiver, one or two letters is sufficient. Common variables such as loop indices and readers can be a single letter (i, r). More unusual things and global variables need more descriptive names.`

Then this debate reduced to whether `info` is unusual or not in the context of this function - which is a 10 line function and the life of this variable is 3-4 line. 

IMO, to learn the meaning of open guidance (like usual v/s unusual), one will have to learn by looking at standard code, beyond guidelines. As an example a [sample code snippet](https://github.com/golang/go/blob/dc50d69119ca8070dd16fb968b46f21f854d89fb/src/cmd/go/internal/list/list.go#L594) from golang src code. This `if` block, (and next `for loop`) we all may find unbearable because local variables are not expressive but that is what is encouraged in this convention.

> If there are two people indicating the readability could be improved, isn't that enough?
It's a matter of choice whether we all collectively stay at what is readable to us or get used to non-expressive local variables of shorter life span?

Yes, this debate has gone overboard which it does not deserve. I'll change this to `info` to keep you both happy but still I would encourage you to look at the usage of shorter/not expressive local variable in standard code - it's full of this.